### PR TITLE
Improve scan perf by re-enable prefetching in ScanNode

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

**Comment by eddyxu** [comment]
Screenshot 2023-01-27 at 12 32 48 PM
 Before vs after

File: rust/benches/scan.rs [hunk 1]
```diff
@@ -15,6 +15,17 @@
 // specific language governing permissions and limitations
 // under the License.
 
+//! Before running the dataset, prepare a "test.lance" dataset, in the
+//! `lance/rust` directly. There is no limitation in the dataset size,
```
Comment (changhiskhan) [inline rust/benches/scan.rs:19]: directory?

File: rust/benches/scan.rs [hunk 2]
```diff
@@ -15,6 +15,17 @@
 // specific language governing permissions and limitations
 // under the License.
 
+//! Before running the dataset, prepare a "test.lance" dataset, in the
```
Comment (changhiskhan) [inline rust/benches/scan.rs:18]: how to prepare though? python? rust script?

**Review by changhiskhan — COMMENTED** [review]
Curious what the perf impact is. Also CI

File: rust/benches/scan.rs (hunk 2 above)
Comment (eddyxu) [reply rust/benches/scan.rs:18]: You should use either python/rust to generate the dataset. It is also a good test for guaranteeing compatibility between implementations.

**Comment by eddyxu** [comment]
> Curious what the perf impact is. Also CI

So full scan of the highly nested dataset, now it is 3-4x faster than previous, and it is about 2x than the C++ implementation.

**Comment by eddyxu** [comment]
> Also CI

It starts to return batch unordered due to prefetch. need to change the test , working on it.

File: rust/benches/scan.rs (hunk 1 above)
Comment (eddyxu) [reply rust/benches/scan.rs]: fixed.

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**